periodic-extent-spec.md

```diff
@@ -0,0 +1,136 @@
+
+Extensions proposes `bbox` field changes.
+
+## Examples
+
+- [Classic STAC Collection extent example](examples/classic.json)
+- [Periodic STAC Collection extent example](examples/periodic.json)
+
+## Schema
+
+- [JSON Schema](json-schema/schema.json)
+
+### Extent Object
+
+The object describes the spatio-temporal extents of the Collection. Both spatial and temporal extents are required to be specified.
+
+| Element  | Type                                              | Description                                                           |
+| -------- | ------------------------------------------------- | --------------------------------------------------------------------- |
+| spatial  | [Spatial Extent Object](#spatial-extent-object)   | **REQUIRED.** Potential *spatial extents* covered by the Collection.  |
+| temporal | [Temporal Extent Object](#temporal-extent-object) | **REQUIRED.** Potential *temporal extents* covered by the Collection. |
+
+#### Spatial Extent Object
+
+The object describes the spatial extents of the Collection.
+
+| Element | Type         | Description                                                          |
+| ------- | ------------ | -------------------------------------------------------------------- |
+| bbox    | \[\[number]] | **REQUIRED.** Potential *spatial extents* covered by the Collection. |
+
+**bbox**: Bounding Boxes of the assets represented by this Collection using either 2D or 3D geometries. Each outer array element can be a separate bounding box, but it is recommended to only use multiple bounding boxes if a union of them would then include a large uncovered area (e.g. the union of Germany and Chile).
+
+The length of the inner array must be 2*n where n is the number of dimensions. The array contains all axes of the southwesterly most extent followed by all axes of the northeasterly most extent specified in Longitude/Latitude or Longitude/Latitude/Elevation based on [WGS 84](http://www.opengis.net/def/crs/OGC/1.3/CRS84). When using 3D geometries, the elevation of the southwesterly most extent is the minimum depth/height in meters and the elevation of the northeasterly most extent is the maximum.
+
+The coordinate reference system of the values is WGS 84 longitude/latitude. Example that covers the whole Earth: `[[-180.0, -90.0, 180.0, 90.0]]`.  Example that covers the whole earth with a depth of 100 meters to a height of 150 meters: `[[-180.0, -90.0, -100.0, 180.0, 90.0, 150.0]]`.
+
+#### Temporal Extent Object
+
+The object describes the temporal extents of the Collection.
+
+| Element  | Type               | Description                                                           |
+| -------- | ------------------ | --------------------------------------------------------------------- |
+| interval | \[\[string\|[Periodic Interval](#periodic-interval)\|null]] | **REQUIRED.** Potential *temporal extents* covered by the Collection. |
```

Comment:
@jisantuc I think that should be exactly this way. Mix should not be allowed; or [string] or [preiodic interval] 
I don't think this schema would validate heterogeneus lists.

Reply:
I think this isn't quite right -- shouldn't it be

```
[[string|null] | [Periodic Interval]]
```

?

The outer array can contain either an array of classic intervals or an array of periodic intervals

Reply:
@jisantuc :+1: makes sense to me, I'll fix it

Reply:
oh, is not it the same thing?

Reply:
@jisantuc it is `oneOf[string, null, periodicSchema]` (talking about JsonSchema)

Reply:
I don't think it's the same thing --

I think it needs to be (in Scala so that we can communicate like _human beings_) `Array[Option[String]] | Array[PeriodicExtent]`, instead of `Array[Option[String] | PeriodicExtent]`. Since periodic extents are prescriptive, the semantics of a mix of periodic and non-periodic extents are _really weird_, and I don't think they should be possible.

Reply:
@jisantuc I guess that is the argue about signatures, I was trying to follow the STAC Spec style to be consistent.

Reply:
Anyway, I'll fix it for clarity :+1:

Reply:
I don't think that's a deliberate style choice, I think it's a regularly occurring oversight. You can see in label at least that it's two types of arrays rather than an array of one thing or another -- https://github.com/stac-extensions/label/#class-object